core/src/main/java/org/apache/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -98,7 +98,7 @@ public InputRowPlusRaw nextRowWithRaw()
     }
   }
 
-  private LineIterator getNextLineIterator()
+  private LineIterator getNextLineIterator() throws IOException
```

Comment:
Since the new version of `LineIterator` supports `try-with-resources`, we should simplify `Closeable ignore2 = lineIterator != null ? lineIterator::close : null` to `Closeable ignore2 = lineIterator` on line #122.

Reply:
Great suggestion, thanks!

Reply:
Updated the PR

Reply:
👍